### PR TITLE
Seed development users in kratos and the Graph API

### PR DESCRIPTION
**Pull request by thehabbos007**
## 🌟 What is the purpose of this PR?

This PR adds a seed step for creating Kratos identities and Graph user entities for two development users, `alice@example.com` and `bob@example.com`.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1202621375435417/1202714661072444/f) _(internal)_

## 🔍 What does this change?

- Add a seed step before starting the backend ensuring the two users exist

## 📜 Does this require a change to the docs?

- Changed readme to reflect how users are seeded now

## 🛡 What tests cover this?

- manual test

## ❓ How to test this?

1.  Checkout the branch / view the deployment
1.  Run external services `yarn external-services up`
1.  Run the frontend/backend `yarn dev` from the root
1.  Observe backend logs. Should contain two lines looking like `Development user available, email = "alice@example.com" password = "password"`
1.  Login with the given user at http://localhost:3000/login

### CODE REVIEW

File: packages/hash/api/src/auth/seed-dev-users.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import { Logger } from "@hashintel/hash-backend-utils/logger";
+import { AxiosError } from "axios";
+
+import { GraphApi } from "../graph";
+import { UserModel } from "../model";
+import { adminKratosSdk, KratosUserIdentityTraits } from "./ory-kratos";
+
+/** @todo shortname and fullName is currenlty unused,
```
Comment (nonparibus) [inline packages/hash/api/src/auth/seed-dev-users.ts:8]:
```suggestion
/** @todo shortname and fullName is currently unused,
```

Comment (thehabbos007) [reply packages/hash/api/src/auth/seed-dev-users.ts:8]: Thanks!

File: packages/hash/api/src/auth/seed-dev-users.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    fullName: "Alice Alison",
+  },
+  {
+    email: "bob@example.com",
+    shortname: "bob",
+    fullName: "Bob Bobson",
+  },
+] as const;
+
+const devPassword = "password";
```
[the remark this answers is not in the document]
Comment (thehabbos007) [reply packages/hash/api/src/auth/seed-dev-users.ts:29]: These are development passwords, we don't want to customize them with env variables or keep them secret

File: packages/hash/api/src/auth/seed-dev-users.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+export const seedDevUsers = async ({
```
Comment (benwerner01) [inline packages/hash/api/src/auth/seed-dev-users.ts:31]: Can we rename this to something like `ensureDevUsersAreSeeded` to make it more explicit that they aren't seeded when this method is called if they already exist?

**Review by benwerner01 — CHANGES_REQUESTED** [review]
Thanks Ahmad, this is working for me locally. Had a few thoughts which would be good to get in before merging.

File: packages/hash/api/src/auth/seed-dev-users.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+import { GraphApi } from "../graph";
+import { UserModel } from "../model";
+import { adminKratosSdk, KratosUserIdentityTraits } from "./ory-kratos";
+
+/** @todo shortname and fullName is currently unused,
+      these need to be added to the UserModel creation step. */
+type DevelopmentUser = {
+  email: string;
+  shortname: string;
+  fullName: string;
```
[the remark this answers is not in the document]
Comment (thehabbos007) [reply packages/hash/api/src/auth/seed-dev-users.ts]: Addressed in 0466f624de12de324ab2d31cc7e06923b40ce2f9

File: packages/hash/api/src/auth/seed-dev-users.ts (hunk 3 above)
Comment (thehabbos007) [reply packages/hash/api/src/auth/seed-dev-users.ts]: Addressed in d69bedb18fea3f7ceb5117c0ae722fd4f9ac96e3

**Review by benwerner01 — APPROVED** [review]

**Pull request merged**